src/nvim/decoration.c

```diff
@@ -118,7 +118,7 @@ void decor_redraw(buf_T *buf, int row1, int row2, DecorInline decor)
 
 void decor_redraw_sh(buf_T *buf, int row1, int row2, DecorSignHighlight sh)
 {
-  if (sh.hl_id || (sh.flags & (kSHIsSign|kSHSpellOn|kSHSpellOff))) {
+  if (sh.hl_id || (sh.url != NULL) || (sh.flags & (kSHIsSign|kSHSpellOn|kSHSpellOff))) {
```

Comment:
Any reason not to add a `kSHIsURL` flag instead of checking `sh.url`? (Sorry for the late review was waiting if bfredl had something to say about it TBH.)

Reply:
I thought about it but decided against it because it's duplicating information, so no need to waste a flag. "Non null URL" _is_ the flag, in a sense.

I am not opposed to it though if there are other reasons to use a flag.

Reply:
Other than these checks being more readable if it just used flags I don't think so. But it's so only checked a few times and `sh.hl_id` is checked directly as well so probably doesn't matter much :)